trieste/bayesian_optimizer.py

```diff
@@ -176,12 +231,16 @@ def optimize(
             acquisition_rule = cast(AcquisitionRule[S, SP], EfficientGlobalOptimization())
 
         models = {tag: create_model(spec) for tag, spec in model_specs.items()}
-        history: List[LoggingState[S]] = []
+        history: List[Record[S]] = []
 
         for step in range(num_steps):
+            if track_state:
+                history.append(Record(datasets, models, acquisition_state))
+
             try:
                 if track_state:
-                    _save_to_history(history, datasets, models, acquisition_state)
+                    models = {tag: gpflow.utilities.deepcopy(m) for tag, m in models.items()}
```

Comment:
You can replace this with just
```suggestion
                    models = {tag: copy.deepcopy(m) for tag, m in models.items()}
```
if you're willing to constraining to tensorflow_probability >=0.12 (i.e. tensorflow >=2.4).

Reply:
i have another PR for this #97